src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1468,6 +1472,109 @@ def create_ocsp_request(self, builder):
         )
         return _OCSPRequest(self, ocsp_req)
 
+    def _create_ocsp_basic_response(self, builder, private_key, algorithm):
+        basic = self._lib.OCSP_BASICRESP_new()
+        self.openssl_assert(basic != self._ffi.NULL)
+        basic = self._ffi.gc(basic, self._lib.OCSP_BASICRESP_free)
+        evp_md = self._lib.EVP_get_digestbyname(
+            builder._response._algorithm.name.encode("ascii")
+        )
+        self.openssl_assert(evp_md != self._ffi.NULL)
+        certid = self._lib.OCSP_cert_to_id(
+            evp_md, builder._response._cert._x509,
+            builder._response._issuer._x509
+        )
+        self.openssl_assert(certid != self._ffi.NULL)
+        certid = self._ffi.gc(certid, self._lib.OCSP_CERTID_free)
+        if builder._response._revocation_reason is None:
+            reason = -1
+        else:
+            reason = _CRL_ENTRY_REASON_ENUM_TO_CODE[
+                builder._response._revocation_reason
+            ]
+        if builder._response._revocation_time is None:
+            rev_time = self._ffi.NULL
+        else:
+            rev_time = self._lib.ASN1_TIME_new()
+            self.openssl_assert(rev_time != self._ffi.NULL)
+            rev_time = self._ffi.gc(rev_time, self._lib.ASN1_TIME_free)
+            self._set_asn1_time(rev_time, builder._response._revocation_time)
```

Comment:
I'm the worst... but let's factor this "new ASN1_time + gc + set asn1time" into a "create_asn1_time" function